lightning/src/ln/channelmanager.rs

```diff
@@ -2731,6 +2738,50 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
+	/// Send a payment that is probing the given route for liquidity. We calculate the
+	/// [`PaymentHash`] of probes based on a static secret and a random [`PaymentId`], which allows
+	/// us to easily discern them from real payments. This can be checked by calling
+	/// [`payment_is_probe`].
+	///
+	/// [`payment_is_probe`]: Self::payment_is_probe
+	pub fn send_probe_payment(&self, route: &Route) -> Result<(PaymentHash, PaymentId), PaymentSendFailure> {
```

Comment:
Yes, @jkczyz had previously brought up MPP probes, too. However,  MPP probes would lead to each path being probed for less than `amount_msat`, which might be confusing to the user. Or we'd enforce that all paths should be probed for the given amount, which also not really makes sense.

So, I'd argue we should just stick to single-path probes, as there is likely not much to be gained from MPP probes in general?

Reply:
Hum, but does that make sense given I'd then need to build a `Route` from scratch to give it to `send_payment_internal`? And I surely don't want to recreate all that logic in another method just so it would take a `&[&RouteHop]`? 

With the refactoring in https://github.com/lightningdevkit/rust-lightning/pull/1567/commits/1eab1855cc4d159099e595782d25756456f8ae4e, I think we should get away with using a `Route`, since we can not only check whether `paths.len() == 1`, but also enforce it by setting `max_path_count = 1`?

Reply:
That said, if you see it as a requirement, I'm of course happy to update it to use `Vec<RouteHop>` and will then see to wrap this into a more streamlined API in the follow-up.

Reply:
Fair enough, will update accordingly.

Reply:
Switched to `Vec<RouteHop>` with 842a321d.